crates/cheatnet/src/runtime_extensions/forge_runtime_extension/cheatcodes/cheat_execution_info.rs

```diff
@@ -0,0 +1,189 @@
+    value: T,
+    span: CheatSpan,
+    target: ContractAddress,
+}
+
+#[derive(Clone, Default, Debug)]
+pub enum Operation<T> {
+    StartGlobal(T),
+    Start(CheatArguments<T>),
+    Stop(ContractAddress),
+    StopGlobal,
+    #[default]
+    Retain,
+}
+
+#[derive(Clone, Default, Debug, PartialEq, Eq)]
+pub struct TxInfoMock {
+    pub version: CheatStatus<Felt252>,
+    pub account_contract_address: CheatStatus<Felt252>,
+    pub max_fee: CheatStatus<Felt252>,
+    pub signature: CheatStatus<Vec<Felt252>>,
+    pub transaction_hash: CheatStatus<Felt252>,
+    pub chain_id: CheatStatus<Felt252>,
+    pub nonce: CheatStatus<Felt252>,
+    pub resource_bounds: CheatStatus<Vec<Felt252>>,
+    pub tip: CheatStatus<Felt252>,
+    pub paymaster_data: CheatStatus<Vec<Felt252>>,
+    pub nonce_data_availability_mode: CheatStatus<Felt252>,
+    pub fee_data_availability_mode: CheatStatus<Felt252>,
+    pub account_deployment_data: CheatStatus<Vec<Felt252>>,
+}
+
+impl TxInfoMock {
+    #[must_use]
+    pub fn is_mocked(&self) -> bool {
+        self != &Default::default()
+    }
+}
+
+#[derive(Clone, Default, Debug, PartialEq, Eq)]
+pub struct BlockInfoMock {
+    pub block_number: CheatStatus<u64>,
+    pub block_timestamp: CheatStatus<u64>,
+    pub sequencer_address: CheatStatus<ContractAddress>,
+}
+
+impl BlockInfoMock {
+    #[must_use]
+    pub fn is_mocked(&self) -> bool {
+        self != &Default::default()
+    }
+}
+
+#[derive(Clone, Default, Debug)]
+pub struct ExecutionInfoMock {
+    pub block_info: BlockInfoMock,
+    pub tx_info: TxInfoMock,
+    pub caller_address: CheatStatus<ContractAddress>,
+    pub contract_address: CheatStatus<ContractAddress>,
+    pub entry_point_selector: CheatStatus<EntryPointSelector>,
+}
+
+#[derive(Clone, Default, Debug)]
+pub struct TxInfoDto {
+    pub version: Operation<Felt252>,
+    pub account_contract_address: Operation<Felt252>,
+    pub max_fee: Operation<Felt252>,
+    pub signature: Operation<Vec<Felt252>>,
+    pub transaction_hash: Operation<Felt252>,
+    pub chain_id: Operation<Felt252>,
+    pub nonce: Operation<Felt252>,
+    pub resource_bounds: Operation<Vec<Felt252>>,
+    pub tip: Operation<Felt252>,
+    pub paymaster_data: Operation<Vec<Felt252>>,
+    pub nonce_data_availability_mode: Operation<Felt252>,
+    pub fee_data_availability_mode: Operation<Felt252>,
+    pub account_deployment_data: Operation<Vec<Felt252>>,
+}
+
+#[derive(Clone, Default, Debug)]
+pub struct BlockInfoDto {
+    pub block_number: Operation<u64>,
+    pub block_timestamp: Operation<u64>,
+    pub sequencer_address: Operation<ContractAddress>,
+}
+
+#[derive(Clone, Default, Debug)]
+pub struct ExecutionInfoDto {
+    pub block_info: BlockInfoDto,
+    pub tx_info: TxInfoDto,
+    pub caller_address: Operation<ContractAddress>,
+    pub contract_address: Operation<ContractAddress>,
+    pub entry_point_selector: Operation<EntryPointSelector>,
+}
+
+macro_rules! for_all_fields {
```

Comment:
Wouldn't keeping those things in a map, and having simple functions (ops) be simpler in terms of code readability?

Reply:
They have different types (not only different felt types) so there would be need for enum of all them and then some kind of unwraping. Sadly it would not be more readable

Reply:
Meh i kind of dislike this macrology for this kind of stuff since adding or modifying of the code is harder
I can live with that though, but would wait for another opinion on that

Reply:
I'd say it's readable enough right now